src/Components/Cards/VerticalCard.jsx

```diff
@@ -8,10 +10,60 @@ export const VerticalCard = ({ videoDetail }) => {
     statistics: { viewCount },
   } = videoDetail;
 
+  const {
+    auth: { token },
+  } = useAuth();
+
+  const { addToWatchlaterOnServer, removeFromWatchlaterOnServer } =
+    useAxiosCalls();
+
+  const { setShowLogin } = useModal();
+
+  const {
+    videoState: { watchlater },
+  } = useVideo();
+  const [watchlaterButton, setWatchlaterButton] = useState("far fa-clock");
+
+  const watchlaterConfig = {
+    url: "/api/user/watchlater",
+    body: { video: { ...videoDetail } },
+    headers: { headers: { authorization: token } },
+  };
+
+  const addToWatchlaterClickHandler = () => {
+    if (token) {
+      addToWatchlaterOnServer(watchlaterConfig);
+      setWatchlaterButton("fas fa-clock");
+    } else {
+      setShowLogin(true);
+    }
+  };
+
+  const removeFromWatchlaterClickHandler = () => {
+    removeFromWatchlaterOnServer(watchlaterConfig);
+    setWatchlaterButton("far fa-clock");
+  };
+
+  const watchlaterButtonStatus = () => {
+    if (watchlaterButton === "far fa-clock") {
+      addToWatchlaterClickHandler();
+    } else {
+      removeFromWatchlaterClickHandler();
+    }
+  };
+
+  useEffect(() => {
+    if (watchlater.findIndex((el) => el._id === videoDetail._id) !== -1) {
+      setWatchlaterButton("fas fa-clock");
+    } else {
+      setWatchlaterButton("far fa-clock");
+    }
+  }, [watchlater, videoDetail._id, setWatchlaterButton]);
```

Comment:
Why you are doing this in use effect you can do it directly.
put a condition to that button if the item in watch later than show the fill icon otherwise the normal one.

Reply:
It is used to check after every render, whether the item is there in the list or not.